opm/autodiff/FlowMain.hpp

```diff
@@ -391,6 +391,8 @@ namespace Opm
                 deck_ = parser->parseFile(deck_filename, parseContext);
                 checkDeck(deck_, parser);
                 eclipse_state_.reset(new EclipseState(deck_, parseContext));
+                auto ioConfig = eclipse_state_->getIOConfig();
+                ioConfig->setOutputDir(output_dir_);
```

Comment:
> There are some things about const and pointers that are not so intuitive,

I don't understand it ...

> That is one reason it would be better to return a reference, that would force the method to be mutable, which makes sense.

I would actually vote for `mutable` here; as a general rule I think the `EclipseState` object should be `const` for the entire lifetime - however there are some aspects which we want to override based on commandline switches which take presedence over the deck, as in this case, since these are non-standard I think it is OK to use `mutable`.

Anyway - I think this particular detail should be fixed later.